### ISSUE TITLE
[Request]  Please update the Android APK

### ISSUE THREAD

**Issue by Sing-Li** [mlc-ai/binary-mlc-llm-libs#10]
The current one is already 2 months old.  Seems the CI/Build process won't do it automatically.

**Comment by BarfingLemurs** [comment]
sharing g-link apk with llama 2 from developer server
https://drive.google.com/file/d/1AlvpiSlCgPSxrMwqLh46ofBg3U1tEueV/view?usp=sharing

**Comment by remixer-dec** [comment]
Thanks for sharing tha APK @BarfingLemurs. I just tried it with 4 different models, including [these](https://github.com/mlc-ai/mlc-llm/commit/b2fbc3c798d2b2843e07e2bf545e1c8b1b735d26), none of them work for me, even RedPajama that worked in the previous version no longer works, after initializing, the chat button is always inactive. Snapdragon / 8GB.

**Comment by BarfingLemurs** [comment]
For me, during the first DL, I couldn't run it, because my phone turned off, and a bad connection. This is the source of most of the bug reports, the first few model parts were unfinished and too small. Look under the app data files from your PC and you may notice some files much smaller than in the repo.

I corrected this error, I was then able to see the completion message after initialization on a 6gb snapdragon phone. I killed a few processes to do this. I couldn't run, but you can.

**Comment by bscott-monrovia** [comment]
That link is working for me. Note 20 Ultra. (Snapdragon 12GB)

**Comment by MrCsabaToth** [comment]
I tried both the apk linked on the site (which points to a 3 month old apk) and this recent apk. The old apk had two models configured: vicuna-v1-7b-q4f16_0 and the RedPajama-INCITE-Chat-3B-v1-q4f16_0: I was also able to open the chat, and got the ready message, but never got an answer. I also downloaded the [rwkv-raven 1b5](https://huggingface.co/mlc-ai/mlc-chat-rwkv-raven-1b5-q8f16_0) and it starts, however the send button never becomes active.

The new apk has Llama-2-7b-chat-hf-q4f16_1 configured. It starts and initializes, but I never get an answer. I downloaded the RedPajama-INCITE-Chat-3B-v1-q4f16_1, however just like with the old APK + rwkv model the chat opens, but the send button never becomes active.

In summary: so far I was not able to chat with only the old Vicuna model.
I tried on two devices:
1. OnePlus Nord EU 12GB RAM, Android 12, Snapdragon 765G 5G (7 nm) w Adreno 620
2. Motorola ThinkPhone 8GB RAM, Android 13, Snapdragon 8+ Gen 1 (4 nm) w Adreno 730

What am I doing wrong?

**Comment by BarfingLemurs** [comment]
@MrCsabaToth I feel I can explain this:

>I also downloaded the [rwkv-raven 1b5](https://huggingface.co/mlc-ai/mlc-chat-rwkv-raven-1b5-q8f16_0) and it starts, however the send button never becomes active.

You can literally only use the two models. The apk itself needs a .jar file compiled during conversion+quantization.

I suspect the reason for the model's publication on HF was as a test. Yes, its labeled for the apk. No, you cannot use the model unless the apk has the .jar file.

**Comment by MrCsabaToth** [comment]
> You can literally only use the two models. The apk itself needs a .jar file compiled during conversion+quantization.
> 
> I suspect the reason for the model's publication on HF was as a test. Yes, its labeled for the apk. No, you cannot use the model unless the apk has the .jar file.
> 

1. One jar for each model?
2. Is that the actual API for the model?
3. Are there instructions on how to produce / come by the jar?
4. Why does the app allow you to specify other models if they won't work?

I was snooping the LogCat and didn't see any crash related to any missing jar or class or stuff like that.

**Comment by BarfingLemurs** [comment]
1. & 3. These instructions show you how to make it. https://mlc.ai/mlc-llm/docs/deploy/android.html Shit, it's .tar, not .jar.
2. There is a CLI executable which some people got running with termux (not me)
4. Likely poc testing, but they never really said other models could work.

**Comment by MrCsabaToth** [comment]
> 1. & 3. These instructions show you how to make it. https://mlc.ai/mlc-llm/docs/deploy/android.html Shit, it's .tar, not .jar.
> 
>     2. There is a CLI executable which some people got running with termux (not me)
> 
>     3. Likely poc testing, but they never really said other models could work.
> 
> 
> fyi side loading the model won't work, you need to download it.
> 
> I'm sorry if the app isn't working/supported but it seemed some others got it working, likely adreno as well, so I don't know much about your issue.

I see "The model lib dist/RedPajama-INCITE-Chat-3B-v1-q4f16_1/RedPajama-INCITE-Chat-3B-v1-q4f16_1-android.tar will be packaged as a static library into the android app. Checkout [Android App](https://mlc.ai/mlc-llm/docs/deploy/android.html#deploy-android) for more details." @ https://mlc.ai/mlc-llm/docs/compilation/compile_models.html

I need to read more or use the models already there. I wonder if those tar files could be available with the models in the Hugging Face repositories, that would make them usable once the app would know the format.

**Comment by remixer-dec** [comment]
FYI: there is a of a forked mlc chat [build](https://github.com/OpenGVLab/OmniQuant/releases/tag/v0.0.1) with new quantization methods, does [not](https://github.com/OpenGVLab/OmniQuant/issues/7#issuecomment-1712567672) work for me but may work for people who were able to run 7B models. It has the same package name, so the original apk+data needs to be deleted/moved.

**Comment by Hzfengsy** [comment]
Updated with commit https://github.com/mlc-ai/binary-mlc-llm-libs/commit/a6b0a4cdaef24725aaf710d4a05ea8b13da2ce42
